prisma/schema.prisma

```diff
@@ -32,12 +32,41 @@ model Team {
   createdAt DateTime @default(now())
   updatedAt DateTime @updatedAt
 
+  competitionTeams CompetitionTeams[]
+
+  @@map("competitions")
+}
+
+model Team {
+  id Int @id @default(autoincrement())
+  title String
+  key String @unique
+  primaryColor String
+  secondaryColor String
+  tertiaryColor String
+
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  competitionTeams CompetitionTeams[]
   teamMembers TeamMember[]
   teamSheets TeamSheet[]
 
   @@map("teams")
 }
 
+model CompetitionTeams {
+  competition Competition @relation(fields: [competitionId], references: [id])
```

Comment:
i think we should add cascade delete (aka dependent delete from rails) to these relations

```
competition Competition @relation(fields: [competitionId], references: [id], onDelete: Cascade)
team Team @relation(fields: [teamId], references: [id], onDelete: Cascade)
```

Reply:
Hmm ok so thinking this through. 

Let's use the team the wallabies as an example.

The Wallabies is a team
The Rugby ChampionShip is a competition
The Rugby World Cup is a competition

The Wallabies play in both of these competitions.

The Wallabies team has 2 competitionTeam records, one for each of these competitions.

Whilst unlikely it is possible for a team to no longer be part of a competition. So in this example lets say the Wallabies are no longer part of the Rugby Championship.

If we delete the competitionTeam record would we also then be deleting the wallabies team and the Rugby Championship competition?

Reply:
its the other way around, if we delete Wallabies we would delete the competitionTeams records that reference wallabies

Reply:
Oh thats right - sick juan